syntax = "proto3";
package collector_job;




import "collector_service_sdk/model/collector_service/collector_template.proto";

/*
ListCollectorJob请求
*/
message ListCollectorJobRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 pageSize = 2 ;
    /*
    过滤条件(按jobName字段模糊匹配）
    */
    string q = 3 ;
    /*
    关联模版id
    */
    string templateInstanceId = 4 ;
    /*
    模型id
    */
    string objectId = 5 ;
}

/*
ListCollectorJob返回
*/
message ListCollectorJobResponse {
message List {
message CollectTime {
    /*
    采集开始时间
    */
    string startTime = 1 ;
    /*
    采集结束时间
    */
    string endTime = 2 ;
}
        
    
    /*
    关联采集模版
    */
    collector_service.CollectorTemplate collectorTemplate = 1 ;
    /*
    实例ID
    */
    string instanceId = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    模型id
    */
    string objectId = 4 ;
    /*
    任务名称
    */
    string jobName = 5 ;
    /*
    过滤条件
    */
    string filter = 6 ;
    /*
    采集时段
    */
    CollectTime collectTime = 7 ;
    /*
    执行超时
    */
    int32 timeout = 8 ;
    /*
    是否启用
    */
    bool enabled = 9 ;
    /*
    采集间隔
    */
    int32 interval = 10 ;
    /*
    关系查询路径
    */
    string path = 11 ;
    /*
    关系查询路径ID
    */
    string pathId = 12 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListCollectorJobApi返回
*/
message ListCollectorJobResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collector_job.ListCollectorJobResponse data = 4 ;
}

